### PR TITLE
[Core] Check that temp_dir must be absolute path.

### LINKED ISSUE #30650: [core] ray.init raises Error when giving _temp_dir argument
### What happened + What you expected to happen

Jupyter Server
python 3.10.6

```
ray.init(num_cpus = 32)
```
works But

```
ray.init(num_cpus=32, _temp_dir =".tmp/ray")
```

raised:
```
Traceback (most recent call last):
  File "python3.10/site-packages/ray/_private/node.py", line 312, in __init__
    ray._private.services.wait_for_node(
  File "python3.10/site-packages/ray/_private/services.py", line 385, in wait_for_node
    raise TimeoutError("Timed out while waiting for node to startup.")
TimeoutError: Timed out while waiting for node to startup.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "python3.10/site-packages/IPython/core/interactiveshell.py", line 3398, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "/tmp/ipykernel_1324842/4179850511.py", line 2, in <cell line: 2>
    ray.init(num_cpus=_UG['cores'], _temp_dir =".tmp/ray")
  File "python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "python3.10/site-packages/ray/_private/worker.py", line 1429, in init
    _global_node = ray._private.node.Node(
  File "python3.10/site-packages/ray/_private/node.py", line 319, in __init__
    raise Exception(
Exception: The current node has not been updated within 30 seconds, this could happen because of some of the Ray processes failed to startup.
```

### Versions / Dependencies

CentOS
Python 3.10.6
Ray 2.1.0
Jupyter Server

### Reproduction script

ray.init(num_cpus= 32, _temp_dir = ".tmp")

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by rynewang**
## Why are these changes needed?

`ray.init` accepts a `_temp_dir` arg. The arg should be absolute path because it makes no sense to have a "relative path" on all nodes. However the code did not check that and proceeds as if it is absolute path. This makes the init code to hang and timeout.

Example issues for treating it as absolute: the `latest_session` symlink links to the temp_dir and if it is relative path it won't work.

## Compatibility Analysis

The call `ray.init()` times out when the arg is relative path, so it never worked and now we forbid it. This won't break any existing usage.

## Related issue number

Closes #30650.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/parameter.py [hunk 1]
```diff
@@ -428,6 +428,9 @@ def build_error(resource, alternative):
                 "serialization. Upgrade numpy if using with ray."
             )
 
+        if self.temp_dir is not None and not os.path.isabs(self.temp_dir):
+            raise ValueError("temp_dir must be absolute path or None.")
```
Comment (cadedaniel) [inline python/ray/_private/parameter.py:432]: Can we add a test for this?

**Review by jjyao — COMMENTED** [review]
Can you add a test?

**Comment by rynewang** [comment]
> Can you add a test?

do you have a suggested place to place the test?

**Review by cadedaniel — COMMENTED** [review]
Nice fix! Can you add `[Core]` prefix to the PR title? a process we follow so that release notes are easier to make

File: python/ray/_private/parameter.py (hunk 1 above)
Comment (rynewang) [reply python/ray/_private/parameter.py:432]: done.

**Comment by rynewang** [comment]
force-pushed because the test commit did not have signed-off-by.

File: python/ray/tests/test_basic.py [hunk 2]
```diff
@@ -1114,6 +1114,12 @@ def test_import_ray_does_not_import_grpc():
     import grpc  # noqa: F401
 
 
+# https://github.com/ray-project/ray/issues/36431
+def test_temp_dir_must_be_absolute(shutdown_only):
+    # This test fails with the flag set to false.
```
Comment (jjyao) [inline python/ray/tests/test_basic.py:1119]: which flag you are referring to?

**Review by jjyao — APPROVED** [review]

File: python/ray/tests/test_basic.py (hunk 2 above)
Comment (rynewang) [reply python/ray/tests/test_basic.py]: it was copy pasted from the other test. Updated the comment.

**Review by cadedaniel — APPROVED** [review]

**Comment by rynewang** [comment]
Looks like the test failures are not related to my change. @jjyao would you mind taking a look and merge?

**Pull request merged**